src/dhcp/server.c

```diff
@@ -456,7 +456,7 @@ static void init_packet(GDHCPServer *dhcp_server, struct dhcp_packet *packet,
 	packet->gateway_nip = client_packet->gateway_nip;
 	packet->ciaddr = client_packet->ciaddr;
 	dhcp_add_option_uint32(packet, DHCP_SERVER_ID,
-						dhcp_server->server_nip);
+						get_be32(&dhcp_server->server_nip));
```

Comment:
I found that my dongle (as sink) can't connect back to miracalcast (as source, not finished yet), so I tcpdumped all packets to and from the p2p interface. It showed dongle were trying to connect to a wrong address (`1.50.168.192`)
```
192.168.50.100.49080 > 1.50.168.192.7236: Flags [S], cksum 0x2508 (correct), seq 4028751266, win 14600, options [mss 1460,sackOK,TS val 4294943537 ecr 0,nop,wscale 5], length 0
```

Then I checked DHCP datagram, it showed that SERVER_ID was also `1.50.168.192`, so, I guess this is the root cause.

`dhcp_server->server_nip` is in BE, but `dhcp_add_option_uint32()` will change its byte order again (LE) before store it. That's the reason why I change its byte order.

I guess the line you mentioned above is also related to this issue. After applied my patch, the SERVER_ID will be in BE instead of LE, so no byte order change needed, you can compare it to `dhcp_server->server_nip` directly (both in BE).